ci/scripts/cpp_build.sh

```diff
@@ -30,7 +30,7 @@ if [ -x "$(command -v git)" ]; then
 fi
 
 # TODO(kszucs): consider to move these to CMake
-if [ ! -z "${CONDA_PREFIX}" ]; then
+if [ ! -z "${CONDA_PREFIX}" && "${ARROW_EMSCRIPTEN:-OFF}" = "OFF" ]; then
```

Comment:
```suggestion
if [ ! -z "${CONDA_PREFIX}" ] && [ "${ARROW_EMSCRIPTEN:-OFF}" = "OFF" ]; then
```

or we need to use `-a` instead of `&&`.